### PR TITLE
Fix vcpkg issues on CI

### PR DESCRIPTION
**Pull request by emabrey**
Nuget package writing was not working correctly due to permissions misconfiguration
vcpkg was broken on macOS 11 due to misconfiguration
vcpkg logs were not being uploaded for each CI run because the cache key was not properly unique
The vcpkg submodule was updated
Change vcpkg caching to only cache the executable

Signed-off-by: Emily Mabrey <emabrey@tenacityaudio.org>

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [ ] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Comment by Be-ing** [comment]
It looks like vcpkg isn't even trying to upload to GitHub Packages :/

**Comment by emabrey** [comment]
> 
> 
> It looks like vcpkg isn't even trying to upload to GitHub Packages :/

This is a PR and not a master push so that is expected. I'm about to add testing change to enable it on this PR to see what the problem is.

File: .github/workflows/cmake_build.yml [hunk 1]
```diff
@@ -246,17 +248,17 @@ jobs:
...
+#      uses: actions/cache@v2
+#      with:
+#        path: |
+#          ${{ github.workspace }}/vcpkg/vcpkg${{ env.EXE_SUFFIX }}
+#          ${{ env.VCPKG_CACHE_PATH }}
+#        key: ${{ matrix.config.id }}-${{ hashFiles('**/vcpkg.json') }}-${{ env.VCPKG_COMMIT }}-${{ github.run_number }}
+#        restore-keys: |
+#          ${{ matrix.config.id }}-${{ hashFiles(' **/vcpkg.json') }}-${{ env.VCPKG_COMMIT }}-
+#          ${{ matrix.config.id }}-${{ hashFiles(' **/vcpkg.json') }}-
+#          ${{ matrix.config.id }}-
```
Comment (Be-ing) [inline .github/workflows/cmake_build.yml]:
```suggestion
    - name: Setup vcpkg executable cache
      uses: actions/cache@v2
      with:
        path: ${{ github.workspace }}/vcpkg/vcpkg${{ env.EXE_SUFFIX }}
        key: ${{ matrix.config.id }}-${{ env.VCPKG_COMMIT }}
```

File: .github/workflows/cmake_build.yml [hunk 2]
```diff
@@ -275,7 +270,7 @@ jobs:
         git clone --recurse-submodules ${{ env.WX_GIT_URL }}
         cd wxWidgets
         git checkout ${{ env.WX_GIT_REF }}
-        cmake -S . -B cmake_build -D CMAKE_BUILD_TYPE=${{ env.CMAKE_BUILD_TYPE }}  -D CMAKE_INSTALL_PREFIX=${{ env.WX_INSTALL_DIR }}
+        cmake -S . -B cmake_build -D CMAKE_BUILD_TYPE=${{ env.CMAKE_BUILD_TYPE }}  -D CMAKE_INSTALL_PREFIX=${{ env.WX_INSTALL_DIR }} -D MACOSX_DEPLOYMENT_TARGET=10.12
```
Comment (Be-ing) [inline .github/workflows/cmake_build.yml]:
```suggestion
        cmake -S . -B cmake_build -D CMAKE_BUILD_TYPE=${{ env.CMAKE_BUILD_TYPE }}  -D CMAKE_INSTALL_PREFIX=${{ env.WX_INSTALL_DIR }} -D CMAKE_OSX_DEPLOYMENT_TARGET=10.12
```
`MACOSX_DEPLOYMENT_TARGET` is the environment variable name. https://cmake.org/cmake/help/latest/envvar/MACOSX_DEPLOYMENT_TARGET.html

**Review by Be-ing — APPROVED** [review]
Network error uploading the 32 bit Windows build artifact is unrelated.

**Comment by emabrey** [comment]
> 
> 
> Network error uploading the 32 bit Windows build artifact is unrelated.

Just confirmed that.

**Review by Be-ing — APPROVED** [review]

**Pull request merged**